### PR TITLE
Update consensus.md to include online meetings

### PR DESCRIPTION
**Pull request by dtig**
The consensus document currently implies that consensus can only be reached at an in-person meeting, this is out of date as we regularly use the same consensus process for online CG, and Subgroup meetings. Updating the document to reflect the current process.

### CODE REVIEW

File: process/consensus.md [hunk 1]
```diff
@@ -9,11 +9,11 @@ Consensus is critical in a standards process:
   over time, break down. Recording dissent is therefore a critical part of
   building consensus.
 
-## In-person meeting consensus
+## Consensus at meetings
 
-For in-person meetings, champions are expected to list points for which they
-will seek consensus in the meeting agenda, and new consensus points can be added
-in-person as the discussion proceeds.
+For in-person, and online meetings, champions are expected to list points for which they
```
Comment (tlively) [inline process/consensus.md:14]:
```suggestion
For in-person and online meetings, champions are expected to list points for which they
```

File: process/consensus.md [hunk 2]
```diff
@@ -23,11 +23,12 @@ Polling consensus is done by the chair:
 2. The chair asks all participants to express their opinion to the question,
    asking in turn whether they are `Strongly For`, `For`, `Neutral`, `Against`,
    or `Strongly Against`. Participants vote for a single option by raising their
-   hand, or abstain entirely. Aggregate votes are recorded by the note-taker.
-3. If deemed relevant, the chair can ask certain participants if their wish to
+   hand, voting on a poll, or in chat for online meetings, or abstain entirely.
```
Comment (tlively) [inline process/consensus.md:26]: How about this wording: "Participants either abstain entirely or vote for a single option by raising their hand if meeting in person or by voting on a virtual poll or commenting in chat if meeting online."

File: process/consensus.md [hunk 3]
```diff
@@ -23,11 +23,12 @@ Polling consensus is done by the chair:
...
+   Aggregate votes are recorded by the note-taker.
+4. If deemed relevant, the chair can ask certain participants if their wish to
```
Comment (jeffparsons) [inline process/consensus.md:28]:
```suggestion
4. If deemed relevant, the chair can ask certain participants if they wish to
```

Comment (tlively) [inline process/consensus.md:28]:
```suggestion
4. If deemed relevant, the chair can ask certain participants if they wish to
```

File: process/consensus.md [hunk 4]
```diff
@@ -23,11 +23,12 @@ Polling consensus is done by the chair:
...
    explain their vote for the note-taker.
-4. The chair determines whether consensus was reached.
-5. A participant may decide to register a formal objection to the decision. An
+5. The chair determines whether consensus was reached.
+6. A participant may decide to register a formal objection to the decision. An
```
Comment (tlively) [inline process/consensus.md:31]: Who is the director mentioned below? Is that the same as our W3C liaison?

File: process/consensus.md [hunk 5]
```diff
@@ -59,7 +60,7 @@ official video calls. In the latter case, the consensus vote must be added to
 the agenda at least 24 hours before the video call is scheduled to begin, except
 in the case of general interest votes moving pre-proposals to phase 1, which can
 be added as the discussion proceeds. Consensus decisions are recorded in meeting
-notes and published just like in-person meeting notes are published.
+notes and published just like meeting notes are published.
```
Comment (tlively) [inline process/consensus.md:63]: Maybe just "Consensus decisions are recorded in the published meeting notes"

File: process/consensus.md [hunk 6]
```diff
@@ -87,20 +90,22 @@ We differentiate the following cases:
    three contributors of different affiliations is acceptable as long as there
    is no objection. Consensus will be deemed to not have been reached if
    interested parties did not sign off. At any point in time a contributor can
-   request that final consensus be delayed to an in-person meeting. In this
+   request that final consensus be delayed to a subgroup, or a CG meeting. In this
```
Comment (tlively) [inline process/consensus.md:93]:
```suggestion
   request that final consensus be delayed to a subgroup or CG meeting. In this
```

File: process/consensus.md (hunk 1 above)
Comment (dschuff) [inline process/consensus.md:14]:
```suggestion
For meetings, champions are expected to list points for which they
```

File: process/consensus.md [hunk 7]
```diff
@@ -87,20 +90,22 @@ We differentiate the following cases:
...
    discussions.
 3. Substantial technical changes or additions are usually carried in their own
    GitHub repository by a champion. It is critical that these proposals be able
    to evolve quickly without much process. Early on in such a proposal's
    lifetime no consensus is needed and a single champion can modify the proposal
    at will. As the proposal matures it is expected that the champion will seek
-   collaborators to form a small group. Gauging consensus in the small group is
-   left up to the champion, with input from the chair. When a proposal is near
-   maturity the champion shall bring it to a meeting and seek wider
+   collaborators to form a subgroup. Gauging consensus in the small group is
```
Comment (tlively) [inline process/consensus.md:101]: There should not be an expectation that every proposal requires forming a formal subgroup.
```suggestion
   collaborators to form a small group. Gauging consensus in the small group is
```

File: process/consensus.md [hunk 8]
```diff
@@ -87,20 +90,22 @@ We differentiate the following cases:
...
+   left up to the champion, gauging consensus in a subgroup is left to the chair,
```
Comment (tlively) [inline process/consensus.md:102]:
```suggestion
   left up to the champion, and gauging consensus in a subgroup is left to the chair
```

File: process/consensus.md (hunk 2 above)
Comment (dschuff) [inline process/consensus.md:26]:
```suggestion
   hand, voting in an automated poll, in chat for online meetings, or abstain entirely.
```

File: process/consensus.md (hunk 8 above)
Comment (tlively) [reply process/consensus.md:102]: Since "small groups" include subgroups, it is ambiguous whether the chair or the champion declares consensus in cases where the small group is also a subgroup. It would be good to clarify that "if the small group is a subgroup, then the chair or co-chairs of that subgroup gauge consensus, and otherwise the champion guages consensus."

File: process/consensus.md (hunk 5 above)
Comment (dschuff) [inline process/consensus.md:63]: looking at this section/paragraph more broadly, this seems to be including full-group VC meetings together with github (and now subgroups), so since we are now basically including VC above, this paragraph should be reworked a little more.

File: process/consensus.md [hunk 9]
```diff
@@ -87,20 +90,22 @@ We differentiate the following cases:
...
+   or co-chairs of that subgroup. If a small group, or a subgroup is unable to reach
+   consensus, consensus can be sought at an in-person, or online CG meeting. 
+   When a proposal is near maturity the champion shall bring it to a meeting and seek wider
    consensus on open design points and contended issues. All decisions made by
-   the small group can be revisited until consensus is reached at an in-person
-   meeting.
+   the small group can be revisited until consensus is reached at an in-person,
+   or online CG meeting.
```
Comment (tlively) [inline process/consensus.md:108]:
```suggestion
   the small group can be revisited until consensus is reached at an in-person
   or online CG meeting.
```

File: process/consensus.md [hunk 10]
```diff
@@ -87,20 +90,22 @@ We differentiate the following cases:
...
    GitHub repository by a champion. It is critical that these proposals be able
    to evolve quickly without much process. Early on in such a proposal's
    lifetime no consensus is needed and a single champion can modify the proposal
    at will. As the proposal matures it is expected that the champion will seek
-   collaborators to form a small group. Gauging consensus in the small group is
-   left up to the champion, with input from the chair. When a proposal is near
-   maturity the champion shall bring it to a meeting and seek wider
+   collaborators to form a subgroup. Gauging consensus in the small group is
+   left up to the champion, gauging consensus in a subgroup is left to the chair,
+   or co-chairs of that subgroup. If a small group, or a subgroup is unable to reach
```
Comment (tlively) [inline process/consensus.md:103]:
```suggestion
   or co-chairs of that subgroup. If a small group is unable to reach
```

File: process/consensus.md [hunk 11]
```diff
@@ -87,20 +90,22 @@ We differentiate the following cases:
...
+   consensus, consensus can be sought at an in-person, or online CG meeting.
```
Comment (dschuff) [inline process/consensus.md:104]:
```suggestion
   consensus, consensus can be sought at an in-person or online CG meeting. 
```

File: process/consensus.md [hunk 12]
```diff
@@ -87,20 +90,22 @@ We differentiate the following cases:
...
-   maturity the champion shall bring it to a meeting and seek wider
+   collaborators to form a subgroup. Gauging consensus in the small group is
+   left up to the champion, gauging consensus in a subgroup is left to the chair,
+   or co-chairs of that subgroup. If a small group, or a subgroup is unable to reach
+   consensus, consensus can be sought at an in-person, or online CG meeting. 
+   When a proposal is near maturity the champion shall bring it to a meeting and seek wider
    consensus on open design points and contended issues. All decisions made by
-   the small group can be revisited until consensus is reached at an in-person
-   meeting.
+   the small group can be revisited until consensus is reached at an in-person,
```
Comment (dschuff) [inline process/consensus.md:107]:
```suggestion
   the small group can be revisited until consensus is reached at an in-person
```

**Review by tlively — COMMENTED** [review]
Lots of little grammar nits and a few more substantive suggestions.

File: process/consensus.md (hunk 6 above)
Comment (dschuff) [inline process/consensus.md:93]:
```suggestion
   request that final consensus be delayed to a subgroup or a CG meeting. In this
```

File: process/consensus.md (hunk 2 above)
Comment (dschuff) [reply process/consensus.md:26]: or perhaps even "if attending in-person/online" since in practice we have both in a single meeting.

File: process/consensus.md (hunk 4 above)
Comment (dschuff) [reply process/consensus.md:31]: No, it's the director of the whole W3C. However the process for WG formal objections has changed (I think https://www.w3.org/2023/Process-20231103/#addressing-fo is the current process). Maybe this can be fixed in a separate change.

File: process/consensus.md (hunk 3 above)
Comment (dtig) [reply process/consensus.md]: This has been fixed already, resolving. Thanks!

File: process/consensus.md (hunk 4 above)
Comment (dtig) [reply process/consensus.md:31]: Yeah, I'd prefer to defer reflecting W3C changes to a separate change, this is only for reflecting our current process in the document.

File: process/consensus.md (hunk 8 above)
Comment (dtig) [reply process/consensus.md]: I hit resolve by accident but, small group consensus especially online (ex: on PRs, issues), is usually up to the champion, and sometimes up to the subgroup chair. Whereas consensus at Subgroup meetings is usually upto the Subgroup chait. Will walk back the edit, and reword this better.

**Comment by dtig** [comment]
> Lots of little grammar nits and a few more substantive suggestions.

Thanks for the review, It looks like you're not a fan of the Oxford comma, I see that other reviewers aren't too so I've accepted the suggestions. :)

File: process/consensus.md [hunk 13]
```diff
@@ -23,11 +23,12 @@ Polling consensus is done by the chair:
 2. The chair asks all participants to express their opinion to the question,
    asking in turn whether they are `Strongly For`, `For`, `Neutral`, `Against`,
    or `Strongly Against`. Participants vote for a single option by raising their
-   hand, or abstain entirely. Aggregate votes are recorded by the note-taker.
-3. If deemed relevant, the chair can ask certain participants if their wish to
+   hand, voting in an automated poll, in chat for online meetings, or abstain entirely.
+   Aggregate votes are recorded by the note-taker.
```
Comment (conrad-watt) [inline process/consensus.md:27]:
```suggestion
   hand, voting in an automated poll, or typing in the chat of an online meeting.
   Aggregate votes are recorded by the note-taker. Any participant may choose to
   abstain instead of indicating a vote.
```

**Review by conrad-watt — APPROVED** [review]
Great clarifications!

**Review by penzn — COMMENTED** [review]
Thank you, a needed change. A nit on scope of subgroups.

File: process/consensus.md [hunk 14]
```diff
@@ -69,7 +70,9 @@ We introduce the following concepts to help the online decision process:
   participate in the WebAssembly Community Group or Working Group and are
   interested in a particular topic.
 * *Small group* is a subset of Community Group and Working Group participants
-  who decide to collaborate on a single targeted proposal.
+  who decide to collaborate on a single targeted proposal, or a group of related
+  proposals. If an official [subgroup](https://github.com/WebAssembly/meetings/blob/main/process/subgroups.md)
+  for the proposal exists, this term would refer to the subgroup.
```
[the remark this answers is not in the document]
Comment (dtig) [reply process/consensus.md:76]: Subgroups do have a broader scope, but I don't think that's necessarily relevant here - for seeking consensus on a single issue, the subgroup, or the small group are equivalent. I've made a note below that a subgroup forum should be used if one exists that is relevant to the proposal.

File: process/consensus.md (hunk 5 above)
Comment (dtig) [reply process/consensus.md]: Fixed, thanks!

File: process/consensus.md (hunk 8 above)
Comment (dtig) [reply process/consensus.md]: I've reworked this paragraph a little bit to include subgroups, ptal.

File: process/consensus.md (hunk 5 above)
Comment (dtig) [reply process/consensus.md]: Could you give me more specifically on what would be good to rework? In general having all polls in video meetings having a 24-hour notice, and being published in the notes seems like good practice. Or do you mean that you prefer that this is moved to the section above that talks about meetings only?

File: process/consensus.md [hunk 15]
```diff
@@ -87,20 +91,23 @@ We differentiate the following cases:
...
    at will. As the proposal matures it is expected that the champion will seek
-   collaborators to form a small group. Gauging consensus in the small group is
-   left up to the champion, with input from the chair. When a proposal is near
-   maturity the champion shall bring it to a meeting and seek wider
-   consensus on open design points and contended issues. All decisions made by
-   the small group can be revisited until consensus is reached at an in-person
-   meeting.
+   collaborators to form a small group, or use the subgroup if one relevant to the
+   proposal exists. Gauging online consensus in the small group is left up to the
+   champion or in some cases the subgroup chair. Gauging consensus at a subgroup meeting is
```
Comment (tlively) [inline process/consensus.md:104]: "in some cases" still seems underspecified. Should we just say that either the champion or relevant subgroup chair can declare consensus for online discussions?

File: process/consensus.md [hunk 16]
```diff
@@ -87,20 +91,23 @@ We differentiate the following cases:
...
+   left to the chair or co-chairs of that subgroup. If a small group is unable to reach
+   consensus online or at a subgroup meeting, consensus can be sought at an in-person
+   or online CG meeting. When a proposal is near maturity the champion shall bring it
+   to a meeting and seek wider consensus on open design points and contended issues.
```
Comment (tlively) [inline process/consensus.md:108]:
```suggestion
   to a meeting and seek wider consensus on open design questions and contentious issues.
```

File: process/consensus.md (hunk 15 above)
Comment (penzn) [reply process/consensus.md:104]: I think the issue is that 'small groups' are not always subgroups, maybe it would be better to say that champion is responsible if it is a subgroup (subgroup maybe takes precedence over 'small group'). TBH, I don't know if we really use 'small group' concept in practice.

File: process/consensus.md [hunk 17]
```diff
@@ -58,8 +60,8 @@ online, either on GitHub repositories under the WebAssembly organization or in
 official video calls. In the latter case, the consensus vote must be added to
 the agenda at least 24 hours before the video call is scheduled to begin, except
 in the case of general interest votes moving pre-proposals to phase 1, which can
-be added as the discussion proceeds. Consensus decisions are recorded in meeting
-notes and published just like in-person meeting notes are published.
+be added as the discussion proceeds. Consensus decisions are recorded in the
```
Comment (dschuff) [inline process/consensus.md:63]: The paragraph above includes references to video call meetings, (as opposed to "in-person meetings" as the preceding "consensus at meetings" section used to talk about). But now video calls are added to the preceding section as official meetings.
So I think this paragraph should now say something like the following:

"It is critical that work progresses between full-CG meetings: agreed-upon
designs need to move forward, and new ideas need to reach some level of maturity
before being discussed in a full meeting. To that end, this group can reach consensus
online, either on GitHub repositories, or in smaller groups."

That leaves out the part about adding votes to the agenda 24 hours in advance. I guess that should move to the "consensus at meetings" section above.

Comment (dschuff) [reply process/consensus.md:63]: Maybe we should even rename this section, since now it covers Github and small groups, but not VC meetings?

**Comment by tlively** [comment]
Going to go ahead and close this now that #1682 has landed.
